rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/parametric/model/ParametricFileFacts.java

```diff
@@ -130,11 +138,21 @@ private void reportTypeCheckerMessages(List<Diagnostic> messages) {
 
         public void invalidate(boolean isClosing) {
             summary.invalidate(isClosing);
+            lastCalculateId.incrementAndGet(); // All scheduled calls to calculate are now stale
         }
 
-        public void calculate() {
-            summary.calculateSummary();
-            summary.getMessages().thenAccept(this::reportTypeCheckerMessages);
+        public void calculate(SummaryCalculator calculator, long delay) {
+            // If no new call to `calculate` has been made after `delay`
+            // milliseconds have passed (i.e., `lastCalculateId` hasn't changed
+            // in the meantime), then run the calculation. Else, ignore and
+            // leave the calculation to the new call.
+            var calculateId = lastCalculateId.incrementAndGet();
+            calculateScheduler.schedule(() -> {
+                if (lastCalculateId.get() == calculateId) {
+                    summary.calculateSummary(calculator);
+                    summary.getMessages().thenAccept(this::reportTypeCheckerMessages);
+                }
+            }, delay, TimeUnit.MILLISECONDS);
```

Comment:
instead of a `long delay` (which doesn't tell us the unit) we can also use `Duration`, that's more semantic.

Reply:
Fixed